pkg/controllers/subnetset/vpc_handler.go

```diff
@@ -0,0 +1,118 @@
+}
+
+func (h *VPCHandler) Create(_ context.Context, e event.CreateEvent, _ workqueue.RateLimitingInterface) {
+	ns := e.Object.GetNamespace()
+	log.Info("creating default Subnetset for VPC", "Namespace", ns, "Name", e.Object.GetName())
+	for name, subnetSetType := range defaultSubnetSets {
+		if err := retry.OnError(retry.DefaultRetry, func(err error) bool {
+			return err != nil
+		}, func() error {
+			list := &v1alpha1.SubnetSetList{}
+			label := client.MatchingLabels{
+				common.LabelDefaultSubnetSet: subnetSetType,
+			}
+			nsOption := client.InNamespace(ns)
+			if err := h.Client.List(context.Background(), list, label, nsOption); err != nil {
+				return err
+			}
+			if len(list.Items) > 0 {
+				// avoid creating when nsx-operator restarted if Subnetset exists.
+				log.Info("default subnetset already exists", common.LabelDefaultSubnetSet, subnetSetType)
+				return nil
+			}
+			obj := &v1alpha1.SubnetSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns,
+					Name:      name,
+					Labels: map[string]string{
+						common.LabelDefaultSubnetSet: subnetSetType,
+					},
+				},
+				Spec: v1alpha1.SubnetSetSpec{
+					AdvancedConfig: v1alpha1.AdvancedConfig{
+						StaticIPAllocation: v1alpha1.StaticIPAllocation{
+							Enable: true,
+						},
+					},
+				},
+			}
+			if err := h.Client.Create(context.Background(), obj); err != nil {
+				return err
+			}
+			return nil
+		}); err != nil {
+			log.Error(err, "failed to create SubnetSet", "Namespace", ns, "Name", name)
+		}
+	}
+}
+
+func (h *VPCHandler) Delete(_ context.Context, e event.DeleteEvent, _ workqueue.RateLimitingInterface) {
+	log.Info("cleaning default Subnetset for VPC", "Name", e.Object.GetName())
+	for _, subnetSetType := range defaultSubnetSets {
+		if err := retry.OnError(retry.DefaultRetry, func(err error) bool {
+			return err != nil
+		}, func() error {
+			label := client.MatchingLabels{
+				common.LabelDefaultSubnetSet: subnetSetType,
+			}
+			nsOption := client.InNamespace(e.Object.GetNamespace())
+			obj := &v1alpha1.SubnetSet{}
+			if err := h.Client.DeleteAllOf(context.Background(), obj, label, nsOption); err != nil {
+				return client.IgnoreNotFound(err)
+			}
+			return nil
+		}); err != nil {
+			log.Error(err, "failed to delete SubnetSet", common.LabelDefaultSubnetSet, subnetSetType)
+		}
+	}
+}
+
+func (h *VPCHandler) Generic(_ context.Context, _ event.GenericEvent, _ workqueue.RateLimitingInterface) {
+	log.V(4).Info("VPC generic event, do nothing")
+}
+
+func (h *VPCHandler) Update(_ context.Context, _ event.UpdateEvent, _ workqueue.RateLimitingInterface) {
+	log.V(4).Info("VPC update event, do nothing")
+}
+
+var VPCPredicate = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool {
+		return true
+	},
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		return false
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		return true
+	},
+	GenericFunc: func(genericEvent event.GenericEvent) bool {
+		return false
+	},
+}
```

Comment:
This file looks not related to the patch's topic...?

Reply:
you're right. it may be introduced by wrong merge operation